tests/testthat/test-hjust-text-anchor.R

```diff
@@ -0,0 +1,115 @@
+context("hjust text anchor")
+
+# demonstration of gradient descent algorithm from animation package
+grad.desc <- function(
+  FUN = function(x, y) x^2 + 2 * y^2, rg = c(-3, -3, 3, 3), init = c(-3, 3),
+  gamma = 0.05, tol = 0.001, gr = NULL, len = 50, nmax = 50) {
+  x <- seq(rg[1], rg[3], length = len)
+  y <- seq(rg[2], rg[4], length = len)
+  contour <- expand.grid(x = x, y = y)
+  contour$z <- as.vector(outer(x, y, FUN))
+  
+  nms = names(formals(FUN))
+  grad = if (is.null(gr)) {
+    deriv(as.expression(body(FUN)), nms, function.arg = TRUE)
+  } else {
+    function(...) {
+      res = FUN(...)
+      attr(res, 'gradient') = matrix(gr(...), nrow = 1, ncol = 2)
+      res
+    }
+  }
+  
+  xy <- init
+  newxy <- xy - gamma * attr(grad(xy[1], xy[2]), 'gradient')
+  z <- FUN(newxy[1], newxy[2])
+  gap <- abs(z - FUN(xy[1], xy[2]))
+  i <- 1
+  while (gap > tol && i <= nmax) {
+    xy <- rbind(xy, newxy[i, ])
+    newxy <- rbind(newxy, xy[i + 1, ] - gamma * attr(grad(xy[i + 1, 1], xy[i + 1, 2]), 'gradient'))
+    z <- c(z, FUN(newxy[i + 1, 1], newxy[i + 1, 2]))
+    gap <- abs(z[i + 1] - FUN(xy[i + 1, 1], xy[i + 1, 2]))
+    i <- i + 1
+    if (i > nmax) warning('Maximum number of iterations reached!')
+  }
+  objective <- data.frame(iteration = 1:i, x = xy[, 1], y = xy[, 2], z = z)
+  invisible(list(contour = contour, objective = objective))
+}
+
+dat <- grad.desc()
+contour <- dat$contour
+objective <- dat$objective
+objective <- ldply(objective$iteration, function(i) {
+  df <- subset(objective, iteration <= i)
+  cbind(df, iteration2 = i)
+})
+objective2 <- subset(objective, iteration == iteration2)
+
+grad.desc.viz <- function(hjust) {
+  if (!hjust %in% c(0, 0.5, 1)) {
+    warning("unsupported value of hjust was provided.")
```

Comment:
the warning should be added to the animint2dir compiler, not to the grad.desc.viz helper function.